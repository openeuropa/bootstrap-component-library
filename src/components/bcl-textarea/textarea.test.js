import { merge, renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";

import demoData from "@openeuropa/bcl-data-textarea/data";

const template = "@oe-bcl/bcl-textarea/textarea.html.twig";
const render = (params) => renderTwigFileAsNode(template, params, true);

describe("OE - textarea", () => {
  test("renders correctly", () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });

  test("renders correctly when required", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, required: true })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly when disabled", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, disabled: true })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly when readonly", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, readonly: true })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly when invalid", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, invalid: true })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly when valid", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, valid: true })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly with floating label", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, floating: true })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly when horizontal", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, horizontal: true })
    ).resolves.toMatchSnapshot();
  });
});
