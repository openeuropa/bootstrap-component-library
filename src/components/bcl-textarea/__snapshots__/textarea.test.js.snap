// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - textarea renders correctly 1`] = `
<jest>
  <label
    class="form-label"
    for="example-textarea-1"
  >
    Example textarea
  </label>
  <textarea
    class="form-control"
    id="example-textarea-1"
    placeholder="Text here"
    rows="4"
  />
  <div
    class="invalid-feedback"
  >
    Invalid feedback example
  </div>
  <div
    class="valid-feedback"
  >
    Valid feedback example
  </div>
  <div
    class="form-text"
  >
    An helper text
  </div>
</jest>
`;

exports[`OE - textarea renders correctly when disabled 1`] = `
<jest>
  <label
    class="form-label"
    for="example-textarea-1"
  >
    Example textarea
  </label>
  <textarea
    class="form-control"
    disabled="true"
    id="example-textarea-1"
    placeholder="Text here"
    rows="4"
  />
  <div
    class="invalid-feedback"
  >
    Invalid feedback example
  </div>
  <div
    class="valid-feedback"
  >
    Valid feedback example
  </div>
  <div
    class="form-text"
  >
    An helper text
  </div>
</jest>
`;

exports[`OE - textarea renders correctly when horizontal 1`] = `
<jest>
  <div
    class="row"
  >
    <label
      class="col-form-label col-sm-2"
      for="example-textarea-1"
    >
      Example textarea
    </label>
    <div
      class="col-sm-10"
    >
      <textarea
        class="form-control"
        id="example-textarea-1"
        placeholder="Text here"
        rows="4"
      />
    </div>
    <div
      class="invalid-feedback"
    >
      Invalid feedback example
    </div>
    <div
      class="valid-feedback"
    >
      Valid feedback example
    </div>
    <div
      class="form-text"
    >
      An helper text
    </div>
  </div>
</jest>
`;

exports[`OE - textarea renders correctly when invalid 1`] = `
<jest>
  <label
    class="form-label"
    for="example-textarea-1"
  >
    Example textarea
  </label>
  <textarea
    class="form-control is-invalid"
    id="example-textarea-1"
    placeholder="Text here"
    rows="4"
  />
  <div
    class="invalid-feedback"
  >
    New invalid feedback
  </div>
  <div
    class="valid-feedback"
  >
    Valid feedback example
  </div>
  <div
    class="form-text"
  >
    An helper text
  </div>
</jest>
`;

exports[`OE - textarea renders correctly when readonly 1`] = `
<jest>
  <label
    class="form-label"
    for="example-textarea-1"
  >
    Example textarea
  </label>
  <textarea
    class="form-control"
    id="example-textarea-1"
    placeholder="Text here"
    readonly="true"
    rows="4"
  />
  <div
    class="invalid-feedback"
  >
    Invalid feedback example
  </div>
  <div
    class="valid-feedback"
  >
    Valid feedback example
  </div>
  <div
    class="form-text"
  >
    An helper text
  </div>
</jest>
`;

exports[`OE - textarea renders correctly when required 1`] = `
<jest>
  <label
    class="form-label"
    for="example-textarea-1"
  >
    Example textarea
  </label>
  <textarea
    class="form-control"
    id="example-textarea-1"
    placeholder="Text here"
    required="true"
    rows="4"
  />
  <div
    class="invalid-feedback"
  >
    Invalid feedback example
  </div>
  <div
    class="valid-feedback"
  >
    Valid feedback example
  </div>
  <div
    class="form-text"
  >
    An helper text
  </div>
</jest>
`;

exports[`OE - textarea renders correctly when valid 1`] = `
<jest>
  <label
    class="form-label"
    for="example-textarea-1"
  >
    Example textarea
  </label>
  <textarea
    class="form-control is-valid"
    id="example-textarea-1"
    placeholder="Text here"
    rows="4"
  />
  <div
    class="invalid-feedback"
  >
    Invalid feedback example
  </div>
  <div
    class="valid-feedback"
  >
    New valid feedback
  </div>
  <div
    class="form-text"
  >
    An helper text
  </div>
</jest>
`;

exports[`OE - textarea renders correctly with floating label 1`] = `
<jest>
  <div
    class="form-floating"
  >
    <textarea
      class="form-control"
      id="example-textarea-1"
      placeholder="Text here"
      rows="4"
    />
    <label
      class="form-label"
      for="example-textarea-1"
    >
      Example textarea
    </label>
    <div
      class="invalid-feedback"
    >
      Invalid feedback example
    </div>
    <div
      class="valid-feedback"
    >
      Valid feedback example
    </div>
    <div
      class="form-text"
    >
      An helper text
    </div>
  </div>
</jest>
`;

exports[`OE - textarea renders correctly with text 1`] = `
<jest>
  <label
    class="form-label"
    for="example-textarea-1"
  >
    Example textarea
  </label>
  <textarea
    class="form-control"
    id="example-textarea-1"
    placeholder="Text here"
    rows="4"
  >
    Sample text presented inside
  </textarea>
  <div
    class="invalid-feedback"
  >
    Invalid feedback example
  </div>
  <div
    class="valid-feedback"
  >
    Valid feedback example
  </div>
  <div
    class="form-text"
  >
    An helper text
  </div>
</jest>
`;
